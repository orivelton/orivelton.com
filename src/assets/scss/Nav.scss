.nav {
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 10;
  
  &__group {
    display: flex;
    margin: 0;
  }

  &__menu {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    background: var(--white);
    max-width: 1100px;
    margin: auto;
    padding: 0 2rem;
  }

  &__link {
    padding: 15px;
    color: var(--black);
    font-weight: 500;
    box-shadow: none;
    &:hover {
      font-weight: bold;
    }

    &--no-padding-right {
      padding-right: 0;
    }
    
    &--active {
      font-weight: bold;
      position: relative;
      &::before {
        content: '';
        width: 0; 
        height: 0; 
        border-left: 5px solid transparent;
        border-right: 5px solid transparent;
        border-bottom: 5px solid --black;
        position: absolute;
        bottom: -7px;
        left: 43%;
      }
    }

    &--logo {
      padding: 0.3rem 0;
    }
  }

}

.logo {
  width: 3rem;
}


