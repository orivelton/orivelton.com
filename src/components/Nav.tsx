import { Box, Flex, HStack, Link } from '@chakra-ui/react'
import { DarkModeSwitch } from './DarkModeSwitch'
import Logo from './Logo'

const Links = ['Projects', 'Blog', 'Contact']


export default function Nav() {

  return (
    <>
      <Box px={4}>
        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
          <HStack spacing={8} alignItems={'center'}>
            <Logo />
          </HStack>
          <Box alignItems={'center'}>
            <HStack
              as={'nav'}
              spacing={4}>
              {Links.map((link) => (
                <Link
                  key={link}
                  px={2}
                  py={1}
                  rounded={'md'}
                  _hover={{ textDecoration: 'none', bg: 'gray.200' }}
                  href={'#'}>
                  {link}
                </Link>
              ))}
            </HStack>
          </Box>
          <DarkModeSwitch />
        </Flex>
      </Box>
    </>
  );
}