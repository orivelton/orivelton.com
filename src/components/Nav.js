import React from 'react'
import Logo from './Logo';
import '../assets/scss/Nav.scss';
import { Link } from 'gatsby'

const Nav = () => (
  <nav className="nav">
    <div className="nav__menu">
        <Logo />
        <ul className="nav__group">
          <li>
            <Link to="/blog" className="nav__link" activeClassName='nav__link--active' title="Got to Blog page">Blog</Link>
          </li>
          <li>
            <Link to="/projects" className="nav__link" activeClassName='nav__link--active' title="Got to Projects page">Projects</Link>
          </li>
          <li>
            <Link to="/contact" className="nav__link nav__link--no-padding-right" activeClassName='nav__link--active' title="Got to Contact page">Contact</Link>
          </li>
        </ul>
    </div>
    <hr />
  </nav>
);

export default Nav;