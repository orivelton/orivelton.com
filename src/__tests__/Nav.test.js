import React from 'react';
import renderer from 'react-test-renderer';
import Nav from '../components/Nav';

describe('src/components/Nav.js', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Nav />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
});
