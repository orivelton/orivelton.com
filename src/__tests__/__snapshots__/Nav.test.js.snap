// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Nav.js renders correctly 1`] = `
<nav
  className="nav"
>
  <div
    className="nav__menu"
  >
    <a
      className="nav__link nav__link--logo"
      href="/"
      title="Go to Home"
    >
      <img
        alt="Logo Orivelton Cesar"
        className="logo"
        src="test-file-stub"
      />
    </a>
    <ul
      className="nav__group"
    >
      <li>
        <a
          className="nav__link"
          href="/blog"
          title="Got to Blog page"
        >
          Blog
        </a>
      </li>
      <li>
        <a
          className="nav__link nav__link--no-padding-right"
          href="/contact"
          title="Got to Contact page"
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
  <hr />
</nav>
`;
